modules/apps/map/map-openstreetmap/src/main/resources/META-INF/resources/js/MapOpenStreetMap.es.js

```diff
@@ -30,17 +32,18 @@ class MapOpenStreetMap extends MapBase {
 	 * @param  {Array} args List of arguments to be passed to State
 	 * @review
 	 */
-	constructor(...args) {
+	constructor(args = {}) {
 		MapBase.DialogImpl = OpenStreetMapDialog;
 		MapBase.GeocoderImpl = OpenStreetMapGeocoder;
 		MapBase.GeoJSONImpl = OpenStreetMapGeoJSON;
 		MapBase.MarkerImpl = OpenStreetMapMarker;
 		MapBase.SearchImpl = null;
 
-		super(...args);
+		super({...args, tileURI});
+
+		this.tileURI = tileURI;
```

Comment:
This is odd because if you ran `new MapOpenStreetMap({titleURI: 'my-tile'})`, it wouldn't actually set here. It would be setting the value in the class it extends from(which from my comment above, shouldn't happen because the parent classes shouldn't be concerned with how other classes extend it)

Reply:
I believe it should look like

```js
	constructor(args = {tileURI: defaultTileURI}) {
		MapBase.DialogImpl = OpenStreetMapDialog;
		MapBase.GeocoderImpl = OpenStreetMapGeocoder;
		MapBase.GeoJSONImpl = OpenStreetMapGeoJSON;
		MapBase.MarkerImpl = OpenStreetMapMarker;
		MapBase.SearchImpl = null;

		super(args);

		const {tileURI} = args;

		this.tileURI = tileURI;
```


By doing this, you can then remove any reference to tileURI in MapBase

Reply:
> This is odd because if you ran new MapOpenStreetMap({titleURI: 'my-tile'}), it wouldn't actually set here. It would be setting the value in the class it extends from(which from my comment above, shouldn't happen because the parent classes shouldn't be concerned with how other classes extend it)

Mhm, I get this part, which is why I found it weird that it didn't work to just omit the whole part from the parent class. That's about where my understanding of classes starts and ends, but since even that fundamental part wasn't working as expected it was hurting my brain 🤯 

Nevertheless, I'll keep trying 💪